impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultArtifactCoordinatesFactory.java

```diff
@@ -26,14 +26,14 @@
 import org.apache.maven.api.services.ArtifactCoordinatesFactoryRequest;
 import org.eclipse.aether.artifact.ArtifactType;
 
-import static org.apache.maven.impl.ImplUtils.nonNull;
+import static java.util.Objects.requireNonNull;
 
 @Named
 @Singleton
 public class DefaultArtifactCoordinatesFactory implements ArtifactCoordinatesFactory {
     @Override
     public ArtifactCoordinates create(@Nonnull ArtifactCoordinatesFactoryRequest request) {
-        nonNull(request, "request");
```

Comment:
this pattern is re-ocurring a lot for defensive programming. Actually the outcome is quite the same, resulting in `NPE`. With the only benefit of giving `NullPointerException("request")` instead of raw `NullPointerException()`.

High invasive on code for little change. Why not giving dedicated abstraction layer for this? Reducing overhead to minimum, getting the message spec for `free`:

<img width="1057" alt="image" src="https://github.com/user-attachments/assets/16096752-7018-4954-a093-49ed6147d365" />

https://projectlombok.org/features/NonNull